src/plugins/starter/starter-tool.tsx

```diff
@@ -0,0 +1,20 @@
+import { observer } from "mobx-react";
+import React from "react";
+import { IToolTileProps } from "../../components/tools/tool-tile";
+import { StarterContentModelType } from "./starter-content";
+import "./starter-tool.scss";
+
+export const StarterToolComponent: React.FC<IToolTileProps> = observer((props) => {
+  const content = props.model.content as StarterContentModelType;
```

Comment:
I'm concerned about the stability of `content` vis-a-vis its reference in `handleChange()`. Historically, we have had to worry about stale closure issues in situations like this because the `model` could be assumed to be stable across renders but the `content` could not be. Recent architectural changes have improved the situation so that the `content` is now more stable than it was, but I don't think we've determined whether the `content` instability has been eliminated entirely. The easy solution in this case would be to move this line inside the `handleChange()` function, but ultimately we're going to want to determine what the right pattern is for tiles moving forward.

Reply:
Yes this is a good thing to discuss.

If the content changes. Mobx will re-render the component.
That should recreate the handleChange function and update the textarea with the new handler. I'd assume that the old change handler would get garbage collected at this point and no longer be attached to the textarea.

I could see there being a brief period when the content changes, before mobx triggers the re-render. In that case the handler would try to update the wrong content object. But I'd think re-render would happen before any user events (onChange). So then it seems unlikely maybe impossible that the old handler could be called.

Reply:
Every time I've tried to reason things out at this level of granularity I've been wrong in some aspect of it (or at least often enough that I don't trust such reasoning). As such, the mindset I've adopted is that React event handler callbacks should be considered asynchronous vis-a-vis the render function, and so you simply can't make assumptions about things that are true when the function is defined still being true when the function is called, unless you've taken steps to assure that that's the case (e.g. `useRef()`, `useCallback()`, etc.).

Reply:
I haven't been burnt by this yet, so I guess I'm still attached to minimizing boiler plate when it doesn't seem necessary. 

I think dereferencing `model.content` only in the callback (even it was wrapped in `useCallback`) would actually cause a stale callback in some sense.  The callback would not be recreated because the component wouldn't be re-rendered. That's because the callback code is not observed by MobX so MobX won't know to re-render (and recreate the callback) when `model.content` changes.  But as you wrote this would actually be OK since if the content of the model was updated this would be picked up when the callback was called.

In this case `model.content` is used by `<textarea value=...>` so the callback will be re-created no matter what. But it is dangerous since removing that value assignment would mean the content property wouldn't be observed anymore.

The `useCallback` would only update during a render. So if it fixes the problem it means the render must be happening. And if one of the dependencies of the useCallback changes, as far as I know, it will return a new instance of the function (so components using it see the change and get re-rendered). I'm not 100% sure, but I pretty sure that changes to a `useCallback` dependency will not trigger a re-render. Unless that dependency is a MobX object and the component is observing. 

So the only thing `useCallback` should do is prevent some unnecessary re-rendering of the `textarea`. It shouldn't help with stale references compared to a function that isn't wrapped in `useCallback`.  What it might help with is reminding us to add any objects as dependencies which means they'd be dereferenced outside of the callback.  But I'm not sure if the linter would make us add `model` or 'model.content`. If it was just `model`, that wouldn't be enough for MobX.

But in any case this example is too simple to provide good fodder for errors. It seems we are covered multiple ways regardless of how we do it.  I do agree it would be good to establish a pattern. So I guess I need to look at a more complex case and see about the need for the `useRef` and `useCallback`.

Reply:
@scytacki I added a comment to the code to keep an eye out for this.